app/features/receive/receive-cashu-token-hooks.tsx

```diff
@@ -0,0 +1,354 @@
+      cannotClaimReason: string;
+    }
+  | {
+      claimableToken: Token;
+      cannotClaimReason: null;
+    };
+
+/**
+ * Takes a token and returns the account that the token is from.
+ * If the account does not exist, we construct and return an account, but we do not store it in the database.
+ */
+export function useGetCashuTokenSourceAccount(token: Token) {
+  const tokenCurrency = tokenToMoney(token).currency;
+  const { data: allAccounts } = useAccounts({ type: 'cashu' });
+  const existingAccount = allAccounts.find(
+    (a) => a.mintUrl === token.mint && a.currency === tokenCurrency,
+  );
+
+  if (existingAccount) {
+    return existingAccount;
+  }
+
+  const { data } = useSuspenseQuery({
+    queryKey: ['token-source-account', token.mint, tokenCurrency],
+    queryFn: async (): Promise<ExtendedCashuAccount> => {
+      const info = await getMintInfo(token.mint);
+      const isTestMint = await checkIsTestMint(token.mint);
+      return {
+        id: '',
+        type: 'cashu',
+        mintUrl: token.mint,
+        createdAt: new Date().toISOString(),
+        name: info?.name ?? 'Unknown Mint',
+        currency: tokenToMoney(token).currency,
+        isTestMint: isTestMint ?? false,
+        version: 0,
+        keysetCounters: {},
+        proofs: [],
+        isDefault: false,
+      };
+    },
+    staleTime: 3 * 60 * 1000,
+    retry: 1,
+  });
+
+  return data;
+}
+
+const getClaimableProofs = (unspentProofs: Proof[], cashuPubKey: string) => {
+  let cannotClaimReason: string | undefined;
+  const claimableProofs = unspentProofs.filter((proof) => {
+    if (isPlainSecret(proof.secret)) {
+      return true;
+    }
+    if (!isP2PKSecret(proof.secret)) {
+      cannotClaimReason = 'This ecash contains an unknown spending condition.';
+      return false;
+    }
+    try {
+      const pubkey = getP2PKPubkeyFromProofs([proof]);
+      return pubkey === null || pubkey === cashuPubKey;
+    } catch {
+      console.warn('Failed to get pubkey from proof', proof);
+      return false;
+    }
+  });
+  if (claimableProofs.length === 0) {
+    return {
+      claimableProofs: null,
+      cannotClaimReason:
+        cannotClaimReason ?? 'You do not have permission to claim this ecash',
+    };
+  }
+  return { claimableProofs, cannotClaimReason: null };
+};
+
+/**
+ * This hook is used to check which proofs in a token are already spent and which
+ * proofs have spending conditions that this user can satisfy.
+ * @param token - The token to receive
+ * @param cashuPubKey - The public key that the the user can provide signatures for
+ * @returns A token with only proofs that can be claimed. If the token cannot be claimed,
+ * the hook will return a reason why and a null token.
+ */
+export function useGetClaimableToken({
+  token,
+  cashuPubKey,
+}: UseGetClaimableTokenProps) {
+  const { data: tokenData } = useSuspenseQuery({
```

Comment:
what is the suspense boundary that will be used when this suspense query is in progress? the one in wallet?